stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/EventHandlerImpl.kt

```diff
@@ -385,29 +406,42 @@ internal class EventHandlerImpl(
     }
 
     private suspend fun handleEventsInternal(events: List<ChatEvent>) {
-        events.sortedBy { it.createdAt }
-        updateOfflineStorageFromEvents(events)
+        val sortedEvents = events.sortedBy { it.createdAt }
+        updateOfflineStorageFromEvents(sortedEvents)
 
         // step 3 - forward the events to the active channels
+        sortedEvents
+            .filterIsInstance<CidEvent>()
+            .let { cidEvents ->
+                // handle CidEvents with non-null CIDs
+                cidEvents.filter { it.cid != null }
+                    .groupBy { it.cid!! }
+                    .forEach {
+                        val (cid, eventList) = it
+                        if (domainImpl.isActiveChannel(cid)) {
+                            domainImpl.channel(cid).handleEvents(eventList)
+                        }
+                    }
 
-        events.filterIsInstance<CidEvent>()
-            .groupBy { it.cid }
-            .filterNot { it.key.isBlank() }
-            .forEach {
-                val (cid, eventList) = it
-                if (domainImpl.isActiveChannel(cid)) {
-                    domainImpl.channel(cid).handleEvents(eventList)
-                }
+                // some CidEvents, like those for marking reads,
+                // can have null channel ids to signal "all channels"
```

Comment:
On that case a new Event needs to be created

Reply:
That seems like the cleanest approach, to have a MarkAllRead event object, also ensures we handle it